### DISCUSSION TITLE
Is there a way to connect Python to Dbeaver?

### DISCUSSION THREAD

**Discussion by yulikayln45** [dbeaver/dbeaver#11168] [General]
I currently work a company where I have a connection with SSH tunneling to a PostgresSQL database hosted in AWS. I have credentials for this.

I connect through Dbeaver where it is very convenient to see the schemas and do queries.

However, if I want to do any kind of analysis I currently need export the query results into a CSV and load these into Python, which is not very convenient to do multiple times.

Can someone please guide me on how to either: [https://trackeasy.fun/usps/](https://trackeasy.fun/usps/)

Connect from Python to Dbeaver to query and analyze the info without downloading it? Or;

Connect directly from Python to the PostgresSQL database with the SSH tunnel.

**Comment by laundmo** [comment]
when dbeaver is running, the ssh tunnel is running machine-wide. you can set your tunnel to use a static port, and connect to localhost:8765 (example port) using any python library of your choice. 

example:
![image](https://user-images.githubusercontent.com/24855949/113736475-6b898900-96fd-11eb-9609-f44a91375825.png)

**Reply by Mandy88088** [reply]
Hi laundmo, I am having almost same problem with @yulikayln45, I can access the PostgresSQL with SSH using dbeaver, but when using python script i cannot connect, would it be possible to use dbeaver to connect my script to PostgresSQL?
